kjourney/src/app/data.service.ts

```diff
@@ -12,4 +14,12 @@ export class DataService {
   public getRoles(): Promise<Role[]> {
     return this.http.get<Role[]>('api/roles').toPromise()
   }
+
+  public getCapabilities(): Promise<Capability[]> {
```

Comment:
the use of `Promises` provides no additional benefit and additional computation converting an observable to a promise 

can you revert to using Observables and subscribe to the observables in the components which inject the DataService